components/servlet-mgt/org.wso2.carbon.identity.servlet.mgt/src/main/java/org/wso2/carbon/identity/servlet/mgt/CustomCXFNonSpringJaxrsServlet.java

```diff
@@ -0,0 +1,662 @@
+               Application app = providerApp.getProvider();
+               JAXRSServerFactoryBean bean = ResourceUtils.createApplication(app, ignoreApplicationPath,
+                       this.getStaticSubResolutionValue(servletConfig), this.isAppResourceLifecycleASingleton(app,
+                               servletConfig), this.getBus());
+               String splitChar = this.getParameterSplitChar(servletConfig);
+               this.setAllInterceptors(bean, servletConfig, splitChar);
+               this.setInvoker(bean, servletConfig);
+               this.setExtensions(bean, servletConfig);
+               this.setDocLocation(bean, servletConfig);
+               this.setSchemasLocations(bean, servletConfig);
+               List<?> providers = this.getProviders(servletConfig, splitChar);
+               bean.setProviders(providers);
+               List<? extends Feature> features = this.getFeatures(servletConfig, splitChar);
+               bean.getFeatures().addAll(features);
+               bean.setBus(this.getBus());
+               bean.setApplicationInfo(providerApp);
+               bean.create();
+           }
+       }
+   }
+
+   protected boolean isIgnoreApplicationPath(ServletConfig servletConfig) {
+
+       String ignoreParam = servletConfig.getInitParameter(IGNORE_APP_PATH_PARAM);
+       return ignoreParam == null || PropertyUtils.isTrue(ignoreParam);
+   }
+
+   protected void createServerFromApplication(ServletConfig servletConfig) throws ServletException {
+
+       Application app = this.getApplication();
+       JAXRSServerFactoryBean bean = ResourceUtils.createApplication(app, this.isIgnoreApplicationPath(servletConfig),
+               this.getStaticSubResolutionValue(servletConfig),
+               this.isAppResourceLifecycleASingleton(app, servletConfig), this.getBus());
+       String splitChar = this.getParameterSplitChar(servletConfig);
+       this.setAllInterceptors(bean, servletConfig, splitChar);
+       this.setInvoker(bean, servletConfig);
+       this.setExtensions(bean, servletConfig);
+       this.setDocLocation(bean, servletConfig);
+       this.setSchemasLocations(bean, servletConfig);
+       List<?> providers = this.getProviders(servletConfig, splitChar);
+       bean.setProviders(providers);
+       List<? extends Feature> features = this.getFeatures(servletConfig, splitChar);
+       bean.getFeatures().addAll(features);
+       bean.setBus(this.getBus());
+       bean.setApplication(this.getApplication());
+       bean.create();
+   }
+
+   protected Application createApplicationInstance(String appClassName, ServletConfig servletConfig) {
+
+       return null;
+   }
+
+   protected ApplicationInfo createApplicationInfo(String appClassName, ServletConfig servletConfig)
+           throws ServletException {
+
+       Application customApp = this.createApplicationInstance(appClassName, servletConfig);
```

Comment:
Since this method is always returning null, we can remove the method and its invocation. And the next null check also can be removed and keep the else condition body only.